corehq/apps/data_interfaces/models.py

```diff
@@ -76,25 +96,44 @@ def _rule_matches_case(self, case, now):
         if case.type != self.case_type:
             return False
 
-        if (case.server_modified_on >
-                (now - timedelta(days=self.server_modified_boundary))):
+        if self.filter_on_server_modified and \
+                (case.server_modified_on > (now - timedelta(days=self.server_modified_boundary))):
             return False
 
         return all([criterion.matches(case, now)
                    for criterion in self.automaticupdaterulecriteria_set.all()])
 
     def apply_actions(self, case):
-        properties = {}
+        cases_to_update = defaultdict(dict)
         close = False
 
+        def _get_case_property_value(current_case, name):
+            return current_case.resolve_case_property(name)[0].value
+
+        def _add_update_property(name, value, current_case):
+            while name.startswith('parent/'):
+                name = name[7:]
+                current_case = current_case.parent
+            cases_to_update[current_case.case_id][name] = value
+
         for action in self.automaticupdateaction_set.all():
             if action.action == AutomaticUpdateAction.ACTION_UPDATE:
-                properties[action.property_name] = action.property_value
+                # break this out as helper function?
```

Comment:
i don't think it's necessary to break it out but i'd prefer if we don't leave this comment if that's what we decide